simon.py

```diff
@@ -0,0 +1,132 @@
+import time
+import os.path
+import json
+import numpy as np
+import pandas as pd
+from sklearn.metrics import f1_score
+
+import sys
+sys.path.append("./simon")
+from Simon import Simon
+from Simon.Encoder import Encoder
+from Simon.DataGenerator import DataGenerator
+
+from d3m_profiler.build_table import get_datasets
+
+MAX_LEN = 20
+MAX_CELLS = 100
+P_THRESHOLD = 0.3
+CHECKPOINT_DIR = 'simon/Simon/pretrained_models/'
+DATASET_DIR = '/users/data/d3m/datasets'
+
+def main(checkpoint, nb_epoch, batch_size, execution_config):
+    if not os.path.isdir(CHECKPOINT_DIR):
+        os.makedirs(CHECKPOINT_DIR)
+
+    raw_data, header, categories = parse_datasets(get_datasets(DATASET_DIR))
+    print(f'Categories: {categories}', flush=True)
+
+    start = time.time()
+
+    encoder = Encoder(categories=categories)
+    encoder.process(raw_data, MAX_CELLS)
+    X, y = encoder.encode_data(raw_data, header, MAX_LEN)
+    encoder_max_cells = encoder.cur_max_cells
+    category_count = y.shape[1] # need to know number of fixed categories to create model
+    print('Number of fixed categories is: {}'.format(category_count))
+
+    classifier = Simon(encoder=encoder)
+    data = classifier.setup_test_sets(X, y)
```

Comment:
@evanphilipsmith We need to make a moderate change here. The splitting of data, whether train/test split or CV, should be performed consistently when comparing SIMON to our model or any other model. We need to make sure that the splits are the same so that the score compared are the same. Can you look making this change?

Reply:
@bjschoenfeld 
SIMON: 60% train, 30% validation set, 10% test set (code [here](https://github.com/NewKnowledge/simon/blob/4123c99004eda92a35ed404bc72274d269bbe7b6/Simon/Simon.py#L101-L103) for reference)
BYU profiler: 67% train, 33% test (code [here](https://github.com/byu-dml/d3m-profiler/blob/30779327fffcb546797c44d30013791bd0b89d4d/example.py#L72) for reference)

I could pretty easily change the way the data is split for SIMON's model to mimic ours, but it seems like the Keras model has some callbacks that are specifically looking at the validation set loss, (see their training function [here](https://github.com/NewKnowledge/simon/blob/4123c99004eda92a35ed404bc72274d269bbe7b6/Simon/Simon.py#L306-L323)). The model crashes when I pass it empty validation sets.

I would assume that early stopping is important to the model's performance. Should I focus my efforts on changing our model to implement a validation set? Or do you think there is another way to adjust SIMON's test/train/validation set split to resemble ours more closely?

Reply:
I was able to use a 67% training set and 33% test set by removing the early stopping callbacks from the Keras model. The results from both versions are below are:
```
                                            F1 macro     F1 macro
                                            (zero-div    (zero-div 
Validation Set?     Accuracy   F1 micro     returns 0)   returns 1)     F1 weighted
            no        0.9928      0.9681        0.4113       0.4113          0.9695
           yes        0.9923      0.9636        0.4122       0.6622          0.9652
```